test/unit/spec/VerifyCustomFactor_spec.js

```diff
@@ -0,0 +1,263 @@
+/* eslint max-params:[2, 16] */
+import { _, $, internal } from 'okta';
+import createAuthClient from 'widget/createAuthClient';
+import Router from 'LoginRouter';
+import Beacon from 'helpers/dom/Beacon';
+import VerifyCustomFactorForm from 'helpers/dom/VerifyCustomFactorForm';
+import Util from 'helpers/mocks/Util';
+import Expect from 'helpers/util/Expect';
+import resAllFactors from 'helpers/xhr/MFA_REQUIRED_allFactors';
+import resNoPermissionError from 'helpers/xhr/NO_PERMISSION_error';
+import resChallengeClaimsProvider from 'helpers/xhr/MFA_CHALLENGE_ClaimsProvider';
+import resSuccess from 'helpers/xhr/SUCCESS';
+import $sandbox from 'sandbox';
+import LoginUtil from 'util/Util';
+const SharedUtil = internal.util.Util;
+const itp = Expect.itp;
+
+Expect.describe('VerifyCustomFactor', function() {
+  function createRouter(baseUrl, authClient, successSpy, settings, startRouter) {
+    const router = new Router(
+      _.extend(
+        {
+          el: $sandbox,
+          baseUrl: baseUrl,
+          authClient: authClient,
+          globalSuccessFn: successSpy,
+        },
+        settings
+      )
+    );
+
+    Util.registerRouter(router);
+    Util.mockRouterNavigate(router, startRouter);
+    return router;
+  }
+
+  function expectError(test, code, message, controller, resError) {
+    expect(test.form.hasErrors()).toBe(true);
+    expect(test.form.errorMessage()).toBe(message);
+    expectErrorEvent(test, code, message, controller, resError);
+  }
+
+  function expectErrorEvent(test, code, message, controller, resError) {
+    expect(test.afterErrorHandler).toHaveBeenCalledTimes(1);
+    expect(test.afterErrorHandler.calls.allArgs()[0]).toEqual([
+      {
+        controller: controller,
+      },
+      {
+        name: 'AuthApiError',
+        message: message,
+        statusCode: code,
+        xhr: resError,
+      },
+    ]);
+  }
+
+  function expectSubtitleToBe(test, desiredSubtitle) {
+    expect(test.form.subtitleText()).toBe(desiredSubtitle);
+  }
+
+  async function setup(res, selectedFactorProps, settings, lastFailedChallengeFactorData, languagesResponse, startRouter) {
+    const setNextResponse = Util.mockAjax();
+    const baseUrl = 'https://foo.com';
+    const authClient = createAuthClient({ issuer: baseUrl, transformErrorXHR: LoginUtil.transformErrorXHR });
+    const successSpy = jasmine.createSpy('success');
+    const afterErrorHandler = jasmine.createSpy('afterErrorHandler');
```

Comment:
Are we using jest to run these tests? If so, let's please use Jest objects instead of Jasmine:

```suggestion
    const successSpy = jest.fn();
    const afterErrorHandler = jest.fn();
```